### DISCUSSION TITLE
Unity WebGL Not supporting  on vercel

### DISCUSSION THREAD

**Discussion by NavdeepYadavVaionex** [vercel/community#4482] [Help]
### Summary

I am using unity webGL inside my react project. When I am putting everything on lcal environment, everything works fine, but when I host it on vercel, unity content is not served. I have tried with brotil compression and without compression but nothing is working for me

### Steps to Reproduce

Host any unity file +react using iframe on vercel

**Comment by srcnalt** [comment]
Hi @NavdeepYadavVaionex,

There is a react package which helps with that, you can give it a try: https://react-unity-webgl.dev

You will need to build the game with compression disabled though (as far as I know)

```ts
"use client"

import React from "react";
import { Unity, useUnityContext } from "react-unity-webgl";

export default function Game() {

  const { unityProvider } = useUnityContext({
    loaderUrl: "game/game.loader.js",
    dataUrl: "game/game.data",
    frameworkUrl: "game/game.framework.js",
    codeUrl: "game/game.wasm",
  });

  return <Unity unityProvider={unityProvider} />
}
```

<img width="170" alt="image" src="https://github.com/vercel/community/assets/3163281/3e632a54-3489-4267-8603-c1705eeaf97b">

**Reply by NavdeepYadavVaionex** [reply]
I have tried with this but its not loading files from local, I am using iframe to load unity webgl on local which is working fine

**Reply by srcnalt** [reply]
Do you get any error messages in the console? Also could you show the folder structure and the code?

**Reply by NavdeepYadavVaionex** [reply]
I am not getting any error message in console,  Please check my folder structure and code
<img width="1440" alt="Screenshot 2023-10-16 at 9 22 33 PM" src="https://github.com/vercel/community/assets/131948612/56c8349c-5634-4746-bd42-497c3b03a973">

**Reply by srcnalt** [reply]
You can remove `https://localhost:3000` from the paths and your loader URL should be `TrackTitans.loader.js`, follow the structure I shared in the example above.

**Reply by NavdeepYadavVaionex** [reply]
I fixed as you mentioned and its working fine on my local dev environment. but my issue is not resolved, as It was working fine on my local dev environment with using iframe and it was not working on vercel. 

I have check the documentation, it says it does not support newer version of unity after 2020
https://react-unity-webgl.dev/docs/getting-started/installation

**Reply by srcnalt** [reply]
It says it supports newer versions 
`Unity version >= 2020.1.0 or above`

Make sure you have relative paths and not `localhost` hardcoded anywhere.
